packages/fluentui/react-builder/src/components/NavigationPanel.tsx

```diff
@@ -0,0 +1,162 @@
+import { JSONTreeElement } from './types';
+import { AccessibilityError } from '../accessibility/types';
+import { AddTabPanel } from './tabPanels/AddTabPanel';
+
+export type NavigationPanelProps = {
+  accessibilityErrors: AccessibilityError[];
+  activeTab: string;
+  jsonTree: JSONTreeElement;
+  onAddComponent?: (uuid: string, where: string) => void;
+  onCloneComponent?: ({ clientX, clientY }: { clientX: number; clientY: number }) => void;
+  onDeleteSelectedComponent?: () => void;
+  onDragStart?: (info: any, e: any) => void;
+  onGoToParentComponent?: () => void;
+  onOpenAddComponentDialog?: (uuid: string, where: string) => void;
+  onMoveComponent?: ({ clientX, clientY }: { clientX: number; clientY: number }) => void;
+  onSelectComponent?: (jsonTreeElement: JSONTreeElement) => void;
+  onSwitchTab?: (tab: any) => void;
+  selectedComponent?: JSONTreeElement;
+};
+
+export const NavigationPanel: React.FunctionComponent<NavigationPanelProps> = (props: NavigationPanelProps) => {
+  const [{ mode }] = useMode();
+
+  const accessErrorLabelStyle = {
+    position: 'relative',
+    right: '8px',
+    top: '-5px',
+    transform: 'rotate(0deg);',
+    border: 'solid 1px white',
+    height: '18px',
+    width: '18px',
+    justifyContent: 'center',
+    alignItems: 'center',
+  };
+
+  return (
+    <div style={{ display: 'flex', minWidth: '1rem', overflow: 'auto' }}>
+      <Menu
+        accessibility={tabListBehavior}
+        vertical
+        styles={({ theme }) => ({
+          background: '#FAF9F8',
+          border: '0px',
+          borderRight: `1px solid ${theme.siteVariables.colorScheme.default.border2}`,
+          borderRadius: '0px',
+          display: 'flex',
+          flexDirection: 'column',
+          width: '3.4rem',
+          transition: 'opacity 0.2s',
+          position: 'relative',
+          padding: '0px',
+          ...(mode === 'use' && {
+            pointerEvents: 'none',
+            opacity: 0,
+          }),
+        })}
+      >
+        <NavBarItem
+          title="Add components"
+          isSelected={props.activeTab === 'add'}
+          icon={<AddIcon size="large" outline />}
+          onClickHandler={() => props.onSwitchTab('add')}
+        />
+
+        <NavBarItem
+          title="Accessibility"
+          isSelected={props.activeTab === 'accessibility'}
+          icon={
+            props.accessibilityErrors.length !== 0 ? (
+              <>
+                {' '}
+                <AccessibilityIcon size="large" />
+                <Label
+                  design={accessErrorLabelStyle}
+                  color={'red'}
+                  content={props.accessibilityErrors.length}
+                  circular
+                  fluid
+                />{' '}
+              </>
+            ) : (
+              <AccessibilityIcon size="large" outline />
+            )
+          }
+          onClickHandler={() => props.onSwitchTab('accessibility')}
```

Comment:
From my understanding, you're saying that it would be better to create the object using useMemo (eg:
const onSwitchTab = useMemo(() =>  props.onSwitchTab('accessibility'))) and pass it to the onClickHandler. Is that correct?